### PR TITLE
368 convert plugin module to obj to class method

### LINKED ISSUE #293: Create interface for "compare_outputs"
# Requested Update

### Description
* Currently, the "compare_outputs.py" module in the main geoips repo contains a number of different functions for comparing different types of outputs.
* This module should be turned into an interface, to allow modularly specifying the various output comparisons.
* This will also allow easily adding more types of output comparisons (ie, pure python based comparisons to remove reliance on imagemagick, as well as non-image based comparisons to reduce reliance on matplotlib/cartopy versions when comparing outputs).
* Currently compare_outputs is not set up as an interface, so there will be some effort required to rearrange it into a format conducive to using as an interface.

### Background and Motivation
*  [ OPTIONAL: Provide additional detailed information and background related to this issue ]

### Alternative Solutions
*  [ OPTIONAL: If appropriate, list alternative fixes that could be considered]
  
### Environment
*  [ IF APPROPRIATE: Corresponding repository versions ]

### Code to demonstrate issue
*  [ IF APPROPRIATE: Can include steps here to identify/demonstrate bug fix or desired feature ]

### Checklist for Completion
* [ ]

### LINKED ISSUE #368: Make interfaces/base.py --> "plugin_module_to_object" a class method.
# Requested Update

### Description
Currently, the function "plugin_module_to_obj" (found in interfaces/base.py), is standalone and is not included as a class function in BaseModuleInterface. This causes unnecessary confusion, and should be attached to that class interface.

### Background and Motivation
The motivation behind this is the BaseYamlInterface, which includes a method "plugin_yaml_to_object" as a class method. This is a better implementation and BaseModuleInterface should follow that.

### Alternative Solutions
Leave it as is, using self as an additional argument to supplement this.

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***unit tests*** added and pass for new/modified functionality
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***integration tests*** added and pass for new/modified functionality
  * [x] None required. (explain if checked, delete line if not checked)
* [x] Required ***documentation*** added for new/modified functionality
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***release notes*** added for new/modified functionality
  * [x] None required. (explain if checked, delete line if not checked)

# Related Issues
fixes #368 
fixes #293 
Branched off of #351 

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh, which will output a 0 upon success.

# Summary

Currently, base.py: plugin_module_to_obj is a standalone function in interfaces/base.py,
which causes unneccesary confusion and should be converted to a class-based function
under BaseModuleInterface. This follows the convention of plugin_yaml_to_obj under
BaseYamlInterface, which is a much better implementation.

   -  modified: geoips/interfaces/base.py

### CODE REVIEW

File: geoips/interfaces/base.py [hunk 1]
```diff
@@ -666,14 +667,14 @@ def get_plugin(self, name):
                 f" attempting to access the correct plugin name)"
             ) from resp
         # Convert the module into an object
-        return plugin_module_to_obj(self, name, module)
+        return self.plugin_module_to_obj(name, module)
```
Comment (mindyls) [inline geoips/interfaces/base.py:670]: Assuming this is what @jsolbrig was looking for,  and the tests still pass,  it looks good to me

**Review by mindyls — APPROVED** [review]
Assuming this is what @jsolbrig was looking for and the tests still pass,  it looks good to me

File: geoips/interfaces/base.py [hunk 2]
```diff
@@ -618,20 +537,102 @@ def __repr__(self):
...
+    #     )
+    #     if obj.interface != self.name:
+    #         raise PluginError(
+    #             f"Plugin 'interface' attribute on '{obj.name}' plugin does not "
+    #             f"match the name of its interface as specified by entry_points."
+    #         )
+    #     return obj
+
+    @classmethod
+    def plugin_module_to_obj(cls, name, module, obj_attrs={}):
```
Comment (jsolbrig) [inline geoips/interfaces/base.py:556]: One small change. Can we make this _plugin_module_to_obj? I'd prefer that this be "private" (as private as anything in Python) and that it match the Yaml version.

Comment (evrose54) [reply geoips/interfaces/base.py]: Done!

**Review by jsolbrig — APPROVED** [review]

**Pull request merged**